### PR TITLE
New i2cdevices

### PR DESCRIPTION
**Pull request by lgywata**
New I2C devices implementing I2CDevice interface. @lucasdemarchi

### CODE REVIEW

File: libraries/AP_Notify/Display_SSD1306_I2C.cpp [hunk 1]
```diff
@@ -22,33 +22,27 @@ static const AP_HAL::HAL& hal = AP_HAL::get_HAL();
 
 bool Display_SSD1306_I2C::hw_init()
 {
-    // display init sequence
-    uint8_t init_seq[] = {0xAE, 0xD5, 0x80, 0xA8, 0x3F, 0xD3,
+    _dev = hal.i2c_mgr->get_device(2, SSD1306_I2C_ADDRESS);
```
Comment (mirkix) [inline libraries/AP_Notify/Display_SSD1306_I2C.cpp]: @lgywata In long term we should move that into a constructor to make it more generic.

**Comment by mirkix** [comment]
`Display_SSD1306_I2C` rework tested with BBBmini. Thanks @lgywata just want to do the same rework.

File: libraries/AP_Notify/Display_SSD1306_I2C.cpp (hunk 1 above)
Comment (lgywata) [inline libraries/AP_Notify/Display_SSD1306_I2C.cpp]: @mirkix thanks for your review and testing! I'll make the change soon!

**Comment by lgywata** [comment]
see #3998

**Pull request closed without merging**